Data/Map/Internal.hs

```diff
@@ -1566,7 +1581,11 @@ splitAt i0 m0
 -- > updateAt (\_ _  -> Nothing)  2    (fromList [(5,"a"), (3,"b")])    Error: index out of range
 -- > updateAt (\_ _  -> Nothing)  (-1) (fromList [(5,"a"), (3,"b")])    Error: index out of range
 
+#if __GLASGOW_HASKELL__ >= 800
+updateAt :: HasCallStack => (k -> a -> Maybe a) -> Int -> Map k a -> Map k a
+#else
 updateAt :: (k -> a -> Maybe a) -> Int -> Map k a -> Map k a
+#endif
```

Comment:
Would it be better to just make this function total by making an update at a bad index do nothing?

Reply:
This is definitely a more interesting question than just where to add `HasCallStack` and might make more sense for a different issue/PR.  Personally, I'm not sure I've ever used the `***At` functions, so I'm not really the target audience to be weighing in on this decision.  Is there a way you could get community feedback about such a change?  Maybe Haskell-cafe?